feat(items): add filtering by title

codeyourfaceoff opened the pull request:
- add regex mongo querying using the query param `"title"` w/ the `$regex` mongo search option
- add case insenstive search using the `$option: 'i'` for mongo's regex
- santize the user input in the query since we're passing it right to the mongo engine

Add product filtering support to our API
Fixes #6

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

vanessa-cooper commented:
Lovely. Let me take a look.

vanessa-cooper reviewed (changes requested):
Some issues need to be fixed before you can merge your PR:
 - Looks like you run npm install, but we're using yarn. Please remove "package-lock.json" from your PR and use yarn instead.

codeyourfaceoff committed:
feat(items): add filtering by title

- add regex mongo querying using the query param `"title"` w/ the `$regex` mongo search option
- add case insenstive search using the `$option: 'i'` for mongo's regex
- santize the user input in the query since we're passing it right to the mongo engine

Add product filtering support to our API
Fixes #6

vanessa-cooper reviewed (approved):
Looking good! Now we can move on to the UI. 
I'll send you more details on Snack.
